File: core/server/project/shine/src/main/java/com/home/shine/support/pool/BytesWriteStreamPool.java
package com.home.shine.support.pool;

import com.home.shine.bytes.BytesWriteStream;
import com.home.shine.control.ThreadControl;
import com.home.shine.global.ShineSetting;
import com.home.shine.thread.AbstractThread;

/** 字节写流池 */
public class BytesWriteStreamPool extends ObjectPool<BytesWriteStream>
{
	public BytesWriteStreamPool()
	{
		super(BytesWriteStream::create);
	}
	
	@Override
	public void back(BytesWriteStream obj)
	{
		if(obj.getBuf().length>ShineSetting.bytesWriteStreamPoolKeepSize)
			return;
		
		super.back(obj);
	}
	
	/** 取一个 */
	public static BytesWriteStream create()
	{
		AbstractThread thread=ThreadControl.getCurrentShineThread();
		
		if(thread!=null)
		{
			return thread.bytesWriteStreamPool.getOne();
		}
		else
		{
			return BytesWriteStream.create();
		}
	}
	
	/** 还一个 */
	public static void release(BytesWriteStream obj)
	{
		AbstractThread thread=ThreadControl.getCurrentShineThread();
		
		if(thread!=null)
		{
			thread.bytesWriteStreamPool.back(obj);
		}
	}
}

File: core/client/game/src/shine/support/pool/BytesWriteStreamPool.cs
namespace ShineEngine
{
	public class BytesWriteStreamPool:ObjectPool<BytesWriteStream>
	{
		private static BytesWriteStreamPool _instance=new BytesWriteStreamPool();

		public BytesWriteStreamPool():base(()=>new BytesWriteStream(),256)
		{

		}

		public override void back(BytesWriteStream obj)
		{
			if(obj.getBuf().Length>ShineSetting.bytesWriteStreamPoolKeepSize)
				return;


			base.back(obj);
		}

		/** 取一个 */
		public static BytesWriteStream create()
		{
			return _instance.getOne();
		}

		public static void release(BytesWriteStream stream)
		{
			_instance.back(stream);
		}
	}
}